assets/wizards/popups/components/lists-control/index.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import AutocompleteTokenField from '../../../../components/src/autocomplete-tokenfield';
+
+export default function ListsControl( { label, help, placeholder, value, onChange, path } ) {
+	return (
+		<AutocompleteTokenField
+			label={ label }
+			help={ help }
+			placeholder={ placeholder }
+			tokens={ value || [] }
+			fetchSuggestions={ async () => {
+				const lists = await apiFetch( {
+					path,
+				} );
+				const values = Array.isArray( lists ) ? lists : Object.values( lists );
+				const suggestions = values.map( item => ( {
+					value: isNaN( parseInt( item.id ) ) ? item.id.toString() : parseInt( item.id ),
+					label: item.title || item.name,
+				} ) );
+
+				return suggestions;
+			} }
+			fetchSavedInfo={ async ids => {
+				const lists = await apiFetch( {
+					path,
+				} );
+				const values = Array.isArray( lists ) ? lists : Object.values( lists );
+				return values
+					.filter( item => ids.includes( item.id ) )
+					.map( item => ( {
+						value: isNaN( parseInt( item.id ) ) ? item.id.toString() : parseInt( item.id ),
+						label: item.title || item.name,
+					} ) );
```

Comment:
Nit: the function passed to `.map` could be named and reused here and in the other prop. This would be more readable.

Reply:
Good idea, added in 6b26f574